modules/ROOT/pages/cryptography-troubleshooting.adoc

```diff
@@ -0,0 +1,96 @@
+= Troubleshoot Cryptography Module
+
+To troubleshoot Cryptography module, become familiar with the information about app logs, PGP protocol, attached and detached signature, and interpreting commonly thrown messages.
+
+== View the App Log
+
+If you encounter problems while running your Mule runtime engine (Mule) app, check the app log as follows:
+
+* If you’re running the app from Anypoint Platform, the output is visible in the Anypoint Studio console window.
+* If you’re running the app using Mule from the command line, the app log is visible in your OS console.
+
+Unless the log file path is customized in the app’s log file `log4j2.xml`, view the app log in the default location `MULE_HOME/logs/<app-name>.log`.
+
+== Enable Cryptography Module Debug Logging
+
+To begin troubleshooting Cryptography Module, enable debug logging to see the exact error messages:
+
+. Access Anypoint Studio and navigate to the *Package Explorer* view.
+. Open your application’s project name.
+. Open the `src/main/resources` path folder.
+. Open the `log4j2.xml` file inside the folder.
+. Add the following line:
++
+`<AsyncLogger name=”com.mulesoft.modules.cryptography" level="DEBUG" />`
+
+[start=6]
+. Save your changes.
+. Click the project name in *Package Explorer* and then click *Run* > *Run As* > *Mule Application*.
+
+== Troubleshoot PGP Encryption and Decryption
+
+During PGP encryption, the sender of the message must encrypt its content using the receiver’s _public key_. So, whenever you want to encrypt messages in your Mule app using someone else’s public key, in the *Crypto Pgp* global configuration add the _receiver_ public keyring file in the *Public keyring* field.
```

Comment:
Do we need to spell out PGP at first use in narrative here?

Reply:
I wonder if this could be just as effective shortened to...:
```suggestion
If you want to encrypt messages in your Mule app using someone else’s public key, add the `_receiver_ public keyring` file in the *Public keyring* field.
```

Reply:
OR
```suggestion
During PGP encryption, the sender of the message must encrypt its content using the receiver’s _public key_. To encrypt messages in your Mule app using someone else’s public key, in the *Crypto Pgp* global configuration, add the `_receiver_ public` keyring file in the *Public keyring* field.
```
(Not sure what part of the end of this sentence is the name of a file, but file names are typically mono.